src/common/middleware/call-api.js

```diff
@@ -50,39 +50,26 @@ export default defaults => () => next => action => {
                 }));
               }
 
-              return Promise.resolve(result);
```

Comment:
After a bit of debugging I found that we had a bit of confusing and unnecessary code in this middleware.

The middleware already **always** returns a promise (because we `return fetch(...)`). So additional return `Promise.resolve()` or `Promise.reject()` was unnecessary.